src/sim/pseudo_inst.hh

```diff
@@ -64,18 +64,29 @@ decodeAddrOffset(Addr offset, uint8_t &func)
     func = bits(offset, 15, 8);
 }
 
+struct GuestAddr
+{
+    uint64_t addr;
+};
```

Comment:
As we agreed this is a temporary patch to the problem, maybe we should add a comment here detailing why this Addr wrapper is needed